superset/connectors/sqla/models.py

```diff
@@ -965,19 +983,47 @@ def get_sqla_query(  # pylint: disable=too-many-arguments,too-many-locals,too-ma
             main_metric_expr, label = literal_column("COUNT(*)"), "ccount"
             main_metric_expr = self.make_sqla_column_compatible(main_metric_expr, label)
 
-        select_exprs: List[Column] = []
-        groupby_exprs_sans_timestamp = OrderedDict()
+        # To ensure correct handling of the ORDER BY labeling we need to reference the
+        # metric instance if defined in the SELECT clause.
+        metrics_exprs_by_label = {
+            m.name: m for m in metrics_exprs  # pylint: disable=protected-access
+        }
 
-        assert extras is not None
+        # Since orderby may use adhoc metrics, too; we need to process them first
+        orderby_exprs: List[ColumnElement] = []
+        for orig_col, ascending in orderby:
+            col: Optional[Union[Metric, ColumnElement]] = orig_col
```

Comment:
Can `col` really be `Optional` in this context?  I thought it's always defined.

Reply:
Yeah, it used to be optional from `adhoc_metric_to_sqla`. I changed `adhoc_metric_to_sqla` to always return a column, but forgot to update here.